apps/ai-image-tagging/frontend/src/components/AppView/AppView.js

```diff
@@ -0,0 +1,171 @@
+      app.isInstalled(),
+      space.getContentTypes()
+    ]);
+
+    const allContentTypesIds = allContentTypes.items.map(({ sys: { id } }) => id);
+
+    // Following eslint error is caused due to using async/await
+    // eslint-disable-next-line react/no-did-mount-set-state
+    this.setState({ isInstalled, allContentTypesIds }, () => app.setReady());
+
+    app.onConfigure(this.installApp);
+    app.onConfigurationCompleted(err => {
+      if (!err) {
+        this.setState({ isInstalled: true });
+      }
+    });
+  }
+
+  installApp = async () => {
+    const { sdk } = this.props;
+    const { isInstalled, allContentTypesIds, contentTypeId, contentTypeName } = this.state;
+
+    if (isInstalled) {
+      sdk.notifier.success('The app is already fully configured.');
+      return false;
+    }
+
+    if (!contentTypeName) {
+      sdk.notifier.error('Provide a name for the content type.');
+      return false;
+    }
+
+    const isContentTypeIdTaken = allContentTypesIds.includes(contentTypeId);
+    if (isContentTypeIdTaken) {
+      sdk.notifier.error(
+        `ID "${contentTypeId}" is taken. Try a different name for the content type`
+      );
+      return false;
+    }
+
+    let contentType = null;
+    try {
+      const data = makeContentType(contentTypeId, contentTypeName);
+      contentType = await sdk.space.createContentType(data);
+    } catch (error) {
+      sdk.notifier.error(`Failed to create content type "${contentTypeName}"`);
+      return false;
+    }
+
+    // Set the newly created content type's state to "Published"
+    try {
+      await sdk.space.updateContentType(contentType);
+    } catch (error) {
+      sdk.notifier.error(`Failed to publish content type "${contentTypeName}"`);
+      return false;
+    }
+
+    return {
+      targetState: {
+        EditorInterface: {
+          [contentType.sys.id]: {
+            controls: [{ fieldId: 'imageTags' }]
+          }
+        }
+      }
+    };
+  };
+
+  onContentTypeNameChange = ({ target: { value } }) =>
+    this.setState(oldState => ({
+      ...(oldState.isContentTypeIdPristine && { contentTypeId: camelCase(value) }),
+      contentTypeName: value
+    }));
+
+  onContentTypeIdChange = ({ target: { value } }) =>
+    this.setState({
+      isContentTypeIdPristine: false,
+      contentTypeId: value
+    });
+
+  render() {
+    const { isInstalled, allContentTypesIds, contentTypeId, contentTypeName } = this.state;
+
+    return (
+      <>
+        <div className={styles.background} />
+        <div className={styles.body}>
+          <Typography>
+            <Heading className={styles.heading}>About AI Image Tagging</Heading>
+            <Paragraph>
+              The AI Image Tagging app uses AWS Rekognition’s AI and deep learning capabilities to automatically assign tags to images. The tags are visible and accessible via the Contentful web app sidebar and tags are searchable in the search bar.
```

Comment:
We shouldn't be mentioning the vendor we use (Rekognition), we're whitelabelling it with this app